src/stream/src/executor_v2/batch_query.rs

```diff
@@ -66,10 +75,39 @@ where
             .await?
         {
             let ops = vec![Op::Insert; data_chunk.cardinality()];
-            let stream_chunk = StreamChunk::from_parts(ops, data_chunk);
+            // Filter out rows
+            let filtered_data_chunk = match self.filter_chunk(data_chunk) {
+                Some(chunk) => chunk,
+                None => {
+                    continue;
+                }
+            };
+
+            let stream_chunk = StreamChunk::from_parts(ops, filtered_data_chunk);
             yield Message::Chunk(stream_chunk);
         }
     }
+
+    /// Now we use hash as a workaround for supporting parallelized chain.
+    fn filter_chunk(&self, data_chunk: DataChunk) -> Option<DataChunk> {
+        let hash_values = data_chunk
+            .get_hash_values(self.info.pk_indices.as_ref(), CRC32FastBuilder)
+            .unwrap();
+        let n = data_chunk.cardinality();
+        let (columns, _visibility) = data_chunk.into_parts();
```

Comment:
can we ensure that data_chunk's visibility is None?

Reply:
AFAIK, `collect_data_chunk` in `CellBasedTableRowIter` will always return a chunk with `None` visibility.

> By the way, there're also some other executors ignoring the visibility. :(

Added `compact` in `execute_inner`

Reply:
pls add an assert that visibility is None.